### PR TITLE
Orchestrator port parametrized in tests implemented

### PR DESCRIPTION
**Pull request by amandarichardsonn**
Created a parameter that is used for the port argument when instantiating an Orchestrator in the test suite.

Originally, the default port parameter was a hardcoded port number. However, if two local tests run on the same system, local Orchestrators will all be launched on the same port which will result in test failure.

I added the method get_test_port() to the WLMUtils class in conftest.py and replaced each original port argument with the new parameter get_test_port(). Now, local Orchestrators will all be launched on different ports rather than the same.

### CODE REVIEW

File: tests/test_launch_errors.py
Comment (al-rigazzi) [inline tests/test_launch_errors.py:42]: Here we should use a different port for the second `Orchestrator`, to make sure it fails for the right reason (i.e. it fails because we do not allow for two `Orchestrator`s to be launched, not because the port is already taken). I would just add 1 to the port returned by `get_test_port()` and maybe we need to make an explicit reference to this in the test docs (or in the test headers).

File: tests/test_orchestrator.py
Comment (al-rigazzi) [inline tests/test_orchestrator.py:45]: This is OK, but at line 54 we test that the db is running at the correct address. That way was OK as long as we were using a fixed port, can you update the `assert` statement so that it uses the right port?

File: conftest.py
Comment (al-rigazzi) [inline conftest.py:26]: Can we add a line about the used port in `print_test_configuration`? And maybe mention that ports `test_port` and `test_port+1` will be used (see another comment in this review)?

**Review by al-rigazzi — CHANGES_REQUESTED** [review]
Good job! A couple of suggestions and changes to make, but all minor.

Also, can you make sure that the tests added with the last merge make use of `test_port`? I think they might need to use that and a couple more ports... I leave it up to you to pick a good strategy and a good way of communicating that to the user in `print_test_configuration`.

File: tests/test_orchestrator.py
Comment (al-rigazzi) [reply tests/test_orchestrator.py:45]: Also, there is a problem with this test, if the orchestrator fails to launch, we don't shut it down. It would be great if you could fix this as part of this ticket. 
We implemented a couple workarounds for cases like this in other tests or you could possibly also use the `local_db` fixture... I leave it up to you!

File: tests/test_smartredis.py
Comment (al-rigazzi) [inline tests/test_smartredis.py:31]: At the beginning of this file, there is a `REDIS_PORT` constant which we can now delete (and use your new utility wherever it appeared).

**Comment by al-rigazzi** [comment]
All looks very good! I think you might just need to resolve all conflicts and then we are ready to merge.

File: smartsim/settings/base.py
Comment (al-rigazzi) [inline smartsim/settings/base.py:370]: The code in this function was updated on `develop`, but it seems that by fixing the conflicts, we reverted to an older version. I'd suggest to checkout the `develop` version, as it is more readable.

**Review by al-rigazzi — COMMENTED** [review]
Good work! Just one comment (and I think I am responsible for the wrong conflict fix). Once that is addressed, we can merge.

**Review by al-rigazzi — APPROVED** [review]
LGTM!

Great stuff!

**Comment by amandarichardsonn** [comment]
Updating `port` argument in summary (#195 )

Replaced the hard coded `port` argument with the get_test_port() method in the WLMUtils class in conftest.py

[ committed by @amandarichardsonn ]
[ reviewed by @al-rigazzi ]

**Pull request merged**
